apps/desktop-wallet/src/index.tsx

```diff
@@ -48,10 +49,13 @@ ReactDOM.render(
         <Router>
           <Suspense fallback="loading">
             <StyleSheetManager shouldForwardProp={shouldForwardProp}>
-              <QueryClientProvider client={queryClient}>
+              <PersistQueryClientProvider
+                client={queryClient}
+                persistOptions={{ persister: tanstackIndexedDBPersister, maxAge: Infinity }}
```

Comment:
My reasoning for using `Inifnity` as the `maxAge`:

Definitions:
- `maxAge`: _The max-allowed age of the persisted cache. If a persisted cache is found that is older than this time, it will be **silently** discarded (defaults to 24 hours)_
- `gctime`: _The time that `inactive` query results will be kept in memory._
- `inactive`: _Query results that have no mount React components to consume them._

Our current setup has this config:
- `maxAge: Infinity`
- `gcTime`: 5 minutes (default)

The docs suggest that the `gcTime` should be `>= maxAge`, otherwise the garbage collector will remove data from the persisted cache before `maxAge` is reached. However, I don't think this recommendation applies to us.

My argument for the above config is that we don't want the persisted cache discarded after a specific period of time. We want to keep the query data **indefinitely** when `latestTxHash` is still the latest one. We only want it to be discarded when data is garbage collected. Data will be garbage collected after being `inactive` for 5 minutes. Those data are mainly the query results that include an old `latestTxHash`. We have no use for that data.

So let's go through an example:
1. User updates their app to a version that includes Tanstack
2. User launches app and unlocks a wallet
3. ALL data is fetched and they get persisted
4. User closes the app
5. User opens the app after 3 days: The app gets hydrated with the persisted data. Almost all data is currently `inactive` since the user hasn't unlocked any wallet.
7. User unlocks a wallet: The rendered React components will now use some of the `inactive` query data which will make them now `active`. Some old query data will remain `inactive`
8. 5 minutes after the app was launched, any `inactive` data will be garbage collected and hence also removed from persisted cache.

There is only one catch:
1 - 6: Same as above
7. User stays in the "welcome" screen for more than 5 minutes: Since there hasn't been any React components that needed any of the re-hydrated `inactive` data, all data will be garbage collected and hence removed from the persisted cache.
8. User unlocks a wallet after 5 minutes: ALL data are fetched from scratch (as our current prod app behaves)

Reasons why user will remain in "welcome" screen for "too long":
1. Well, because they want to
2. Because of the auto-lock mechanism

Maybe we should set the `gcTime` higher. I would not set it to 24 days though which is the maximum because then it will never run (unless a user keeps their computer on and the app running for more than 24 days). I would also not set it to 1 day. Maybe 1 hour?

---

PS: `stateTime` is irrelevant for this discussion. For for the sake of completion:
- `staleTime`: The time after which the data fetched by the query is considered `stale`. Before that, data is considered `fresh`.

Ref:
- https://tanstack.com/query/latest/docs/framework/react/guides/important-defaults
- https://tanstack.com/query/latest/docs/framework/react/plugins/persistQueryClient

Reply:
To extend the above "catch" case: If a user unlocks wallet A, then locks it, then unlocks wallet B, and keeps the app oper for longer than our `gcTime` (5 minutes), then all persisted cached data of wallet A will be deleted.

In other words: persisted data of a locked wallet get deleted after the `gcTime` (as well as any "inactive" query data of the unlocked wallet).

However, if we set `gcTime` higher, say 1 hour, then only users who keep the app running for longer than 1 hour will have their inactive data cleaned.

Optimally, what we would want is:
1. Persisted cached data of unlocked wallets to NOT be affected by `gcTime` (aka `gcTime: Infinity`)
1. Small `gcTime` data of unlocked wallet for as long as the wallet is unlocked.